src/main/accumulMod.F90

```diff
@@ -595,7 +595,7 @@ subroutine update_accum_field_timeavg(this, level, nstep, field)
 
     do k = begi,endi
        effective_nstep = nstep - this%ndays_reset_shifted(k,level)
-       time_to_reset = (mod(effective_nstep,this%period) == 1 .or. this%period == 1) .and. effective_nstep /= 0
+       time_to_reset = mod(effective_nstep,this%period) == 1 .or. this%period == 1
```

Comment:
I'm pretty sure this change should be reverted. `effective_nstep` refers not to the number of timesteps since run start/restart, but rather to the number of timesteps _since the accumulator field was initialized/reset_.

However, before you revert the change, could you check whether the unit tests in `test_accumul.pf` pass? If so, I think that indicates a gap in the testing coverage.

Reply:
Thank you @samsrabin I will revert the change.
I need to request a tutorial on running test_accumul.pf.